### DISCUSSION TITLE
Introduce a More Expressive Syntax for Database Column Aliasing

### DISCUSSION THREAD

**Discussion by obrunopolo** [laravel/framework#48881] [Ideas]
In the current implementation, using column aliases in database queries involves utilizing the `DB::raw()` method. While this works, it introduces raw SQL into the query, which can be less readable and expressive.

##### Current Syntax:
```php
Post::get(['id', DB::raw('title as name')]);
```

##### Proposed Syntax:
I propose introducing a more intuitive and expressive way to alias database columns directly within the query array, thereby eliminating the need for `DB::raw()` when doing simple column aliasing.

```php
Post::get(['id', 'title' => 'name']);
```

This proposed syntax would streamline code by making it easier to read and write, particularly when dealing with multiple aliases or complex queries. It would align well with Laravel's philosophy of clean, elegant code and could potentially be a backwards-compatible change.

#### Benefits:
- Improves code readability
- Aligns with Laravel's expressive syntax
- Makes the codebase more consistent by minimizing the use of `DB::raw()` for simple tasks

#### Possible Implementation:
A change in the internal query builder to interpret associative array keys as aliases when building the SQL query.

**Comment by shaedrich** [comment]
I'd like to have that. However, knowing Taylor Otwell, it probably won't happen.

**Comment by obrunopolo** [comment]
I'll go further. Maybe this could be extended to joins too.

### Direct Alias in `join()` Method

Currently, joins in Laravel might look like this:

```php
DB::table('orders')
    ->join('users', 'users.id', '=', 'orders.user_id')
    ->select('orders.id', DB::raw('users.name as user_name'))
    ->get();
```

With the extended syntax, it could look like:

```php
DB::table('orders')
    ->join('users', 'users.id', '=', 'orders.user_id')
    ->select(['orders.id', 'users.name' => 'user_name'])
    ->get();
```

**Reply by shaedrich** [reply]
It'd be even nicer if the join method would accept an optional table alias
